OrbitClientModel/SamplingDataPostProcessor.cpp

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitClientModel/SamplingDataPostProcessor.h"
+
+#include <cstdint>
+#include <cstring>
+#include <map>
+#include <memory>
+#include <set>
+#include <vector>
+
+#include "OrbitClientData/Callstack.h"
+#include "OrbitClientData/CallstackTypes.h"
+#include "OrbitClientData/PostProcessedSamplingData.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_set.h"
+#include "capture_data.pb.h"
+
+using orbit_client_protos::CallstackEvent;
+using orbit_client_protos::FunctionInfo;
+using orbit_client_protos::LinuxAddressInfo;
+
+namespace sampling_data_post_processor {
+
+namespace {
+
+class SamplingDataPostProcessor {
+ public:
+  explicit SamplingDataPostProcessor() = default;
+  SamplingDataPostProcessor& operator=(const SamplingDataPostProcessor& other) = default;
+  SamplingDataPostProcessor(const SamplingDataPostProcessor& other) = default;
+
+  SamplingDataPostProcessor(SamplingDataPostProcessor&& other) = default;
+  SamplingDataPostProcessor& operator=(SamplingDataPostProcessor&& other) = default;
+
+  PostProcessedSamplingData ProcessSamples(const CallstackData& callstack_data,
+                                           const CaptureData& capture_data, bool generate_summary) {
+    // Unique call stacks and per thread data
+    callstack_data.ForEachCallstackEvent(
+        [this, &callstack_data, generate_summary](const CallstackEvent& event) {
+          CHECK(callstack_data.HasCallStack(event.callstack_hash()));
```

Comment:
I guess CHECK is declared in OrbitBase/Logging.h, should it be included?

Reply:
It should be included, thanks!